src/ascii_char.rs

```diff
@@ -512,150 +495,125 @@ impl AsciiExt for Ascii {
     }
 }
 
-impl<'a> AsciiCast<'a> for u8 {
-    type Target = Ascii;
-
-    #[inline]
-    unsafe fn to_ascii_nocheck(&self) -> Ascii {
-        transmute(*self)
-    }
-}
-
-impl<'a> AsciiCast<'a> for char {
-    type Target = Ascii;
-
-    #[inline]
-    unsafe fn to_ascii_nocheck(&self) -> Ascii {
-        (*self as u8).to_ascii_nocheck()
-    }
-}
 
-
-/// Error returned by `IntoAscii`.
+/// Error returned by `ToAsciiChar`.
 #[derive(PartialEq)]
-pub struct IntoAsciiError(());
+pub struct ToAsciiCharError(());
 
 const ERRORMSG_CHAR: &'static str = "not an ASCII character";
 
-impl fmt::Debug for IntoAsciiError {
+impl fmt::Debug for ToAsciiCharError {
     fn fmt(&self,  fmtr: &mut fmt::Formatter) -> fmt::Result {
         write!(fmtr, "{}", ERRORMSG_CHAR)
     }
 }
 
-impl fmt::Display for IntoAsciiError {
+impl fmt::Display for ToAsciiCharError {
     fn fmt(&self,  fmtr: &mut fmt::Formatter) -> fmt::Result {
         write!(fmtr, "{}", ERRORMSG_CHAR)
     }
 }
 
-impl Error for IntoAsciiError {
+#[cfg(not(feature = "no_std"))]
+impl Error for ToAsciiCharError {
     fn description(&self) -> &'static str {
         ERRORMSG_CHAR
     }
 }
 
 
-/// Convert `char`, `u8` and other character types to `Ascii`.
-pub trait IntoAscii : AsciiExt {
-    /// Convert to `Ascii` without checking that it is an ASCII character.
-    unsafe fn into_ascii_unchecked(self) -> Ascii;
-    /// Convert to `Ascii`.
-    fn into_ascii(self) -> Result<Ascii,IntoAsciiError>;
+#[cfg(not(feature = "no_std"))]
+/// Convert `char`, `u8` and other character types to `AsciiChar`.
+pub trait ToAsciiChar : AsciiExt {
```

Comment:
For a single char the is_ascii check is easy enough, therefore we can omit the `AsciiExt` bound. Implementers of this trait already check this on their own.